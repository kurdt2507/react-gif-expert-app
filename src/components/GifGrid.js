import React from 'react';
import { useFetchGifs } from '../hooks/useFetchGifs';
import { GifGridItem } from './GifGridItem';

export const GifGrid = ({ category }) => {
    const { data:imagenes, loading } = useFetchGifs(category);

    return (
        <>
            <h3>{ category }</h3>

            { loading && <p>Loading...</p>}

            <div className="card-grid animate__animated animate__fadeIn">
                { imagenes.map((img) => (
                    <GifGridItem
                        key={ img.id }
                        { ...img }
                    />
                )) }
            </div>
        </>
    );
}
