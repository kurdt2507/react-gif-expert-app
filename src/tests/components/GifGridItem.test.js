import { shallow } from 'enzyme';
import { GifGridItem } from '../../components/GifGridItem';

describe('Pruebas de componente <GifGridItem />', () => {
    const title = 'My fake image';
    const url = 'https://localhost/fake_image.png';
    const wrapper = shallow(<GifGridItem title={ title } url={ url } />);

    test('El componente debe coincidir con el snapshot', () => {
        expect(wrapper).toMatchSnapshot();
    });

    test('Valida que el pÃ¡rrafo coincida con el titulo recibido', () => {
        const p = wrapper.find('p').text().trim();

        expect(p).toBe(title);
    });

    test('Verifica que la imagen tenga un source y title y que coincidan con los props recibidos', () => {
        const img = wrapper.find('img');
        const { src, alt } = img.props();
        
        expect(src).toBe(url);
        expect(alt).toBe(title);
    });
    
    test('Comprueba que el div tenga la clase "animate__flipInY"', () => {
        const div = wrapper.find('div');
        const _class = div.prop('className');
        
        expect(_class.includes('animate__flipInY')).toBe(true);
    });
    
});
