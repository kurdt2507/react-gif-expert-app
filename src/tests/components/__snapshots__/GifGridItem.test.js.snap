// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas de componente <GifGridItem /> El componente debe coincidir con el snapshot 1`] = `
<div
  className="card animate__animated animate__flipInY"
>
  <img
    alt="My fake image"
    src="https://localhost/fake_image.png"
  />
  <p>
    My fake image
  </p>
</div>
`;
